### PR TITLE
XW-2774 local nav restyled

### PR DESCRIPTION
**Pull request by idradm**
https://wikia-inc.atlassian.net/browse/XW-2774

### CODE REVIEW

File: extensions/wikia/PremiumPageHeader/PremiumPageHeaderController.class.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+			[ 'title' => 'Videos', 'tracking' => 'explore-videos' ],
+			[ 'title' => 'Images', 'tracking' => 'explore-images' ]
+		];
+
+		return [
+			'text' => 'Explore',
+			'children' => array_map( function ( $page ) {
+				$title = Title::newFromText( $page['title'], NS_SPECIAL );
+				return [
+					'text' => $title->getText(),
```
Comment (latata) [inline extensions/wikia/PremiumPageHeader/PremiumPageHeaderController.class.php]: Do we need both text and textEscaped here?

File: extensions/wikia/PremiumPageHeader/scripts/banner-notifications-hack.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+require(['wikia.window', 'jquery'], function (window, $) {
+	'use strict';
+
+	function moveBannerNotifications() {
+		var $globalNav = $('.wds-global-navigation-wrapper'),
+			$notificationsWrapper = $('.banner-notifications-wrapper');
+
+		$notificationsWrapper.detach();
```
Comment (latata) [inline extensions/wikia/PremiumPageHeader/scripts/banner-notifications-hack.js]: We don't need detach here. InsertAfter detaches element.

File: extensions/wikia/PremiumPageHeader/styles/_local-navigation.scss [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+		}
+	}
+
+	&.pph-local-nav-l3 {
+		box-sizing: border-box;
+		left: 100%;
+		min-width: 100%;
+		min-height: -moz-calc(100% + 1px);
+		min-height: -webkit-calc(100% + 1px);
+		min-height: calc(100% + 1px);
```
Comment (latata) [inline extensions/wikia/PremiumPageHeader/styles/_local-navigation.scss]: All browsers that we support, support `calc` without prefix http://caniuse.com/#feat=calc

File: skins/oasis/modules/templates/Body_Index.php [hunk 4]
```diff
@@ -33,6 +33,12 @@
 	<div id="WikiaPageBackground" class="WikiaPageBackground"></div>
 	<div class="WikiaPageContentWrapper">
 		<?= $app->renderView( 'BannerNotifications', 'Confirmation' ) ?>
+
+		<!-- fixme How should it work with BannerNotification - currently it doesn't look good -->
```
Comment (latata) [inline skins/oasis/modules/templates/Body_Index.php]: remove this comment ;)

File: extensions/wikia/PremiumPageHeader/PremiumPageHeaderController.class.php [hunk 5]
```diff
@@ -0,0 +1,56 @@
+<?php
+
+class PremiumPageHeaderController extends WikiaController {
+
+	public function wikiHeader() {
+		$themeSettings = new ThemeSettings();
+		$settings = $themeSettings->getSettings();
+
+		$this->setVal( 'wordmarkText', $settings["wordmark-text"] );
```
Comment (hakubo) [inline extensions/wikia/PremiumPageHeader/PremiumPageHeaderController.class.php]: ' or " ?

File: extensions/wikia/PremiumPageHeader/PremiumPageHeaderController.class.php [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+			[ 'title' => 'Random', 'tracking' => 'explore-random' ],
+			[ 'title' => 'Community', 'tracking' => 'explore-community' ],
+			[ 'title' => 'Videos', 'tracking' => 'explore-videos' ],
+			[ 'title' => 'Images', 'tracking' => 'explore-images' ]
+		];
+
+		return [
+			'text' => wfMessage( 'pph-explore' )->escaped(),
+			'children' => array_map( function ( $page ) {
+				$title = Title::newFromText( $page['title'], NS_SPECIAL );
```
Comment (hakubo) [inline extensions/wikia/PremiumPageHeader/PremiumPageHeaderController.class.php]: I am pretty sure newFromText can return null, are we afraid of it ? :)

File: extensions/wikia/PremiumPageHeader/PremiumPageHeaderController.class.php [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+		$this->setVal( 'mainPageURL', Title::newMainPage()->getLocalURL() );
+	}
+
+	public function navigation() {
+		$this->setVal( 'data', ( new NavigationModel() )->getLocalNavigationTree( NavigationModel::WIKI_LOCAL_MESSAGE ) );
+		$this->setVal( 'explore', $this->getExplore() );
+		$this->setVal( 'discuss', $this->getDiscuss() );
+	}
+
+	private function getExplore() {
```
Comment (hakubo) [inline extensions/wikia/PremiumPageHeader/PremiumPageHeaderController.class.php:22]:
```php
private function getExplore(): array { 
```
? worth it?

File: extensions/wikia/PremiumPageHeader/scripts/local-navigation.js [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+	'use strict';
+
+	var _track = tracker.buildTrackingFunction({
+		category: 'premium-page-header',
+		trackingMethod: 'analytics'
+	});
+
+	function initTabletSupport() {
+		$('.pph-local-nav-container').on('mouseenter', function () {
+			var container = $(this);
```
Comment (hakubo) [inline extensions/wikia/PremiumPageHeader/scripts/local-navigation.js]: $container?

File: extensions/wikia/PremiumPageHeader/scripts/local-navigation.js [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+
+			// execute this code after all mouse events are done, required for tablets support
+			setTimeout(function () {
+				$('.pph-local-nav-item-l2 > .pph-click').removeClass('pph-click');
+				container.children('a').addClass('pph-click');
+			}, 0);
+		});
+
+		$('.pph-local-nav-item-l1').on('mouseleave', function () {
+			var menuItem = $(this);
```
Comment (hakubo) [inline extensions/wikia/PremiumPageHeader/scripts/local-navigation.js]: $menuItem?

Comment (hakubo) [reply extensions/wikia/PremiumPageHeader/scripts/local-navigation.js]: or
$(this).children('a').removeClass('pph-click')

File: extensions/wikia/PremiumPageHeader/scripts/local-navigation.js [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+			var data = $(element).data('tracking');
+			if (data) {
+				_track({
+					action: tracker.ACTIONS.CLICK,
+					label: data
+				});
+			}
+		};
+
+		$('.pph-local-nav-container > a').on('click', function (e) {
```
Comment (hakubo) [inline extensions/wikia/PremiumPageHeader/scripts/local-navigation.js]: var $this = $(this);

then you can pass it to track as well, so you don't have to wrap it twice

Comment (hakubo) [inline extensions/wikia/PremiumPageHeader/scripts/local-navigation.js]: event!

File: extensions/wikia/PremiumPageHeader/scripts/local-navigation.js [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+				track(this);
+			}
+		});
+		$('.pph-local-nav-tracking:not(.pph-local-nav-container) a').on('click', track);
+	}
+
+	$(function () {
+		initTracking();
+		initTabletSupport();
+	})
```
Comment (hakubo) [inline extensions/wikia/PremiumPageHeader/scripts/local-navigation.js]: omg omg omg, missing ; !

File: extensions/wikia/PremiumPageHeader/scripts/wiki-header.js [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+			});
+		});
+
+		$wikiHeader.find('.pph-add-new-page').on('click', function () {
+			track({
+				action: tracker.ACTIONS.CLICK,
+				label: 'add-new-page'
+			});
+		});
+
```
Comment (hakubo) [inline extensions/wikia/PremiumPageHeader/scripts/wiki-header.js]: I think we need to build linter for JS, 
needless \n

File: extensions/wikia/PremiumPageHeader/styles/_banner-notifications-hack.scss [hunk 13]
```diff
@@ -0,0 +1,25 @@
+@import 'skins/oasis/css/core/breakpoints-variables';
+@import 'skins/oasis/css/core/layout';
+
+.banner-notifications-wrapper {
+	position: relative;
+	z-index: $zTop;
+}
+
+@media #{$breakpoint-small-only} {
```
Comment (hakubo) [inline extensions/wikia/PremiumPageHeader/styles/_banner-notifications-hack.scss]: you could swap it, so you only write class once

```scss
.banner-notifications-wrapper {
  @media #{$breakpoint-small-only} {
     width: $page-width-small - 2 * $content-border-width;
  }
@media #{$breakpoint-medium-only} {
     width: $page-width-medium;
  }
...
 }
```

File: extensions/wikia/PremiumPageHeader/styles/_local-navigation.scss [hunk 14]
```diff
@@ -0,0 +1,197 @@
...
+.pph-local-nav {
+	bottom: 0;
+	left: 290px;
+	position: absolute;
+}
+
+.pph-local-nav-menu {
+	display: flex;
+
+	.pph-local-nav-item-l1 {
```
Comment (hakubo) [inline extensions/wikia/PremiumPageHeader/styles/_local-navigation.scss]: does it have to be nested?

File: extensions/wikia/PremiumPageHeader/styles/_local-navigation.scss [hunk 15]
```diff
@@ -0,0 +1,197 @@
...
+			font-weight: normal;
+		}
+	}
+}
+
+.pph-local-nav-discuss-text:hover {
+	text-decoration: underline;
+}
+
+@media #{$breakpoint-small-only} {
```
Comment (hakubo) [inline extensions/wikia/PremiumPageHeader/styles/_local-navigation.scss]: this could be moved to .pph-local-nav to keep that all in one place

File: extensions/wikia/PremiumPageHeader/styles/_wiki-header.scss [hunk 16]
```diff
@@ -0,0 +1,118 @@
...
+	}
+}
+
+.pph-tally-area {
+	align-items: flex-start;
+	display: flex;
+	font-weight: bold;
+	margin: 20px;
+
+	.pph-tally {
```
Comment (hakubo) [inline extensions/wikia/PremiumPageHeader/styles/_wiki-header.scss]: have to be  nested?

File: extensions/wikia/PremiumPageHeader/styles/_wiki-header.scss [hunk 17]
```diff
@@ -0,0 +1,118 @@
...
+		}
+	}
+
+	svg {
+		fill: $color-button-text;
+		margin-right: 6px;
+	}
+}
+
+@media #{$breakpoint-medium-only} {
```
Comment (hakubo) [inline extensions/wikia/PremiumPageHeader/styles/_wiki-header.scss]: same as above in scss you can 'nest' media to keep all in one place

File: extensions/wikia/PremiumPageHeader/PremiumPageHeaderController.class.php (hunk 7 above)
Comment (idradm) [reply extensions/wikia/PremiumPageHeader/PremiumPageHeaderController.class.php:22]: What does it do? I'm not familiar with that notation.

**Review by latata — APPROVED** [review]

File: extensions/wikia/AssetsManager/config.php [hunk 18]
```diff
@@ -3197,3 +3197,21 @@
 		'//extensions/wikia/ArticleVideo/scripts/article-video.js',
 	],
 ];
+
+$config['premium_page_header_scss'] = [
```
Comment (rybmat) [inline extensions/wikia/AssetsManager/config.php:3201]: I hope that when we productize it, we'll drop this 'premium' prefix 😉

File: extensions/wikia/PremiumPageHeader/PremiumPageHeaderController.class.php (hunk 7 above)
Comment (rybmat) [inline extensions/wikia/PremiumPageHeader/PremiumPageHeaderController.class.php]: it just marks what type is returned by method

File: extensions/wikia/PremiumPageHeader/scripts/banner-notifications-hack.js [hunk 19]
```diff
@@ -0,0 +1,27 @@
...
+			var $notificationsWrapper = $('.banner-notifications-wrapper');
+
+			$notificationsWrapper.insertAfter($globalNav);
+
+			return this;
+		};
+	}
+
+	$(function () {
+		// FIXME: run it only if experiment is active
```
Comment (rybmat) [inline extensions/wikia/PremiumPageHeader/scripts/banner-notifications-hack.js:24]: what about this fixme?
